RoaringBitmap/src/main/java/org/roaringbitmap/art/Node48.java

```diff
@@ -143,6 +143,9 @@ public int getNextSmallerPos(int pos) {
     if (pos == ILLEGAL_IDX) {
       pos = 256;
     }
+    if (pos == 0) {
```

Comment:
pos is not illegal while it's 0. Keep compatible with `getChildKey`

Reply:
Ah damnit. You see what I mean? I made exactly the mistake I literally just warned about. I'm also an idiot :D

Reply:
But then again, what *should* it return? There is no smaller index to return. And returning `0` again doesn't seem right. It's not a `getNextSmallerOrEqualPos` method.

Reply:
suggest `if (pos > 0){pos--;}` or constraint the following `i` with a largest bound.

Reply:
that is `for (; i >=0 && i <32;i--)`

Reply:
You will also note that `getChildKey` will  benefit the performance from  current return `position` logic.

Reply:
> that is `for (; i >=0 && i <32;i--)`

This has exactly the same effect as returning `ILLEGAL_INDEX` directly, since the loop will be skipped.



> suggest `if (pos > 0){pos--;}` 

I'm not convinced this is correct. Pretty sure is would implement `getNextSmallerOrEqualPos`.